File: app/src/main/java/com/templates/kotlintemplates/fragments/BaseFragment.java
package com.templates.kotlintemplates.fragments;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;



public abstract class BaseFragment extends DebugFragment {

    private static final String TAG = BaseFragment.class.getSimpleName();

    protected abstract int getLayoutResource();
    protected View fragmentView;


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        fragmentView = inflater.inflate(getLayoutResource(),container,false);
        return fragmentView;
    }

    public Context getContext() {
        return this.getContext();
    }


    protected void log(String msg) {
        Log.d(TAG, msg);
    }

    protected void toast(String msg) {
        Toast.makeText(getContext(), msg, Toast.LENGTH_SHORT).show();
    }

    protected void toast(int msg) {
        Toast.makeText(getContext(), msg, Toast.LENGTH_SHORT).show();
    }

    protected void snack(View view, int msg, Runnable runnable) {
        this.snack(view, this.getString(msg), runnable);
    }

    protected void snack(View view, int msg) {
        this.snack(view, this.getString(msg), (Runnable) null);
    }

    protected void snack(View view, String msg) {
        this.snack(view, msg, (Runnable) null);
    }

    protected void snack(View view, String msg, final Runnable runnable) {
        Snackbar.make(view, msg, Snackbar.LENGTH_INDEFINITE).setAction("Ok", new View.OnClickListener() {
            public void onClick(View v) {
                if (runnable != null) {
                    runnable.run();
                }
            }
        }).show();
    }

}

File: app/src/main/java/com/templates/kotlintemplates/fragments/BaseFragment.kt
package com.templates.kotlintemplates.fragments

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast


abstract class BaseFragment : DebugFragment() {

    protected abstract val rootLayout: Int
    protected abstract var rootView: View?


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater?.inflate(rootLayout, container, false)
        return rootView
    }

    override fun getContext(): Context {
        return this.context
    }


    override fun log(msg: String) {
        Log.d(TAG, msg)
    }

    protected fun toast(msg: String) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
    }

    protected fun toast(msg: Int) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
    }

    protected fun snack(view: View, msg: Int, runnable: Runnable) {
        this.snack(view, this.getString(msg), runnable)
    }

    protected fun snack(view: View, msg: Int) {
        this.snack(view, this.getString(msg), null as Runnable)
    }

    protected fun snack(view: View, msg: String) {
        this.snack(view, msg, null as Runnable)
    }

    protected fun snack(view: View, msg: String, runnable: Runnable?) {
        Snackbar.make(view, msg, Snackbar.LENGTH_INDEFINITE).setAction("Ok") {
            runnable?.run()
        }.show()
    }

    companion object {

        private val TAG = BaseFragment::class.java.simpleName
    }

}
